docs/security/how-to-authenticate.md

```diff
@@ -0,0 +1,173 @@
+| `gravitino.authenticator.oauth.tokenPath`         | The path for token of the default OAuth server.                                                     | (none)            | Yes if use `oauth` as the authenticator    | 0.3.0         |
+| `gravitino.authenticator.kerberos.principal`      | Indicates the Kerberos principal to be used for HTTP endpoint. Principal should start with `HTTP/`. | (none)            | Yes if use `kerberos` as the authenticator | 0.4.0         |
+| `gravitino.authenticator.kerberos.keytab`         | Location of the keytab file with the credentials for the principal.                                 | (none)            | Yes if use `kerberos` as the authenticator | 0.4.0         |
+
+The signature algorithms that Gravitino supports follows:
+
+| Name  | Description                                    |
+|-------|------------------------------------------------|
+| HS256 | HMAC using SHA-25A                             |
+| HS384 | HMAC using SHA-384                             |
+| HS512 | HMAC using SHA-51                              |
+| RS256 | RSASSA-PKCS-v1_5 using SHA-256                 |
+| RS384 | RSASSA-PKCS-v1_5 using SHA-384                 |
+| RS512 | RSASSA-PKCS-v1_5 using SHA-512                 |
+| ES256 | ECDSA using P-256 and SHA-256                  |
+| ES384 | ECDSA using P-384 and SHA-384                  |
+| ES512 | ECDSA using P-521 and SHA-512                  |
+| PS256 | RSASSA-PSS using SHA-256 and MGF1 with SHA-256 |
+| PS384 | RSASSA-PSS using SHA-384 and MGF1 with SHA-384 |
+| PS512 | RSASSA-PSS using SHA-512 and MGF1 with SHA-512 |
+
+### Example
+
+You can follow the steps to set up an OAuth mode Gravitino server.
+
+1. Prerequisite
+
+   You need to install the JDK8 and Docker.
+
+2. Set up an external OAuth 2.0 server
+
+   There is a sample-authorization-server based on [spring-authorization-server](https://github.com/spring-projects/spring-authorization-server/tree/1.0.3).
+
+   The image has registered client information in the external OAuth 2.0 server.
+
+   Its clientId is `test`. Its secret is `test`. Its scope is `test`.
+
+```shell
+ docker run -p 8177:8177 --name sample-auth-server -d datastrato/sample-authorization-server:0.3.0
+```
+
+3. Open [the JWK URL of the Authorization server](http://localhost:8177/oauth2/jwks) in the browser and you can get the JWK.
+
+   ![jks_response_image](assets/jks.png)
+
+4. Convert the JWK to PEM. You can use the [online tool](https://8gwifi.org/jwkconvertfunctions.jsp#google_vignette) or other tools.
+
+   ![pem_convert_result_image](assets/pem.png)
+
+5. Copy the public key and remove the character `\n` and you can get the default signing key of Gravitino server.
+
+6. You can refer to the [Configurations](gravitino-server-config.md) and append the configurations to the conf/gravitino.conf.
+
+```text
+gravitino.authenticator = oauth
```

Comment:
I remember this key has been replaced with `gravitino.authenticators`, Am I correct?

Reply:
We don't merge this document.

Reply:
I can't get your word `We don't merge this document.`

Reply:
https://github.com/apache/gravitino/pull/4489 This pull request.